class/app.rb

```diff
@@ -0,0 +1,101 @@
+require_relative 'classroom'
+require_relative 'book'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def self.list_books
+    Book.all.map { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+  end
+
+  def self.list_persons
+    Person.all.map { |person| puts "ID: #{person.id} , Name: #{person.name}, Age: #{person.age}" }
+  end
+
+  def self.create_classroom(label)
+    Classroom.new(label)
+  end
+
+  def self.create_person
+    print ' Do you want to create a student (1) or a teacher (2) ? [input the number] :'
+    person_type = gets.chomp.to_i
+    case person_type
+    when 1
+      create_student
+    when 2
+      create_teacher
+    else
+      puts 'Invalid choice. Please enter a valid option.'
+      execute(3)
+    end
+    puts 'Person created successfully'
+  end
+
+  def self.create_student
+    print 'Age: '
+    age = gets.chomp.to_i
+    print 'Name: '
+    name = gets.chomp.to_s
+    print 'Has parent permission ? [Y/N]: '
+    choice = gets.chomp
+    if %w[Y y].include?(choice)
+      Student.new(age, name, parent_permission: true)
+    elsif %w[N n].include?(choice)
+      Student.new(age, name, parent_permission: false)
+    end
+  end
+
+  def self.create_teacher
+    print 'Age: '
+    teacher_age = gets.chomp.to_i
+    print 'Name: '
+    teacher_name = gets.chomp.to_s
+    print 'Specialization: '
+    specialization = gets.chomp.to_s
+    Teacher.new(specialization, teacher_age, teacher_name)
+  end
+
+  def self.create_book
+    print 'Title: '
+    title = gets.chomp.to_s
+    print 'Author: '
+    author = gets.chomp.to_s
+    Book.new(title, author)
+    puts 'Book created successfully'
+  end
+
+  def self.create_rental
```

Comment:
* Kindly fix this error i get in my console when i try to create rentals
![image](https://github.com/HtetWaiYan7191/school-library/assets/52497246/8f3bda09-68ad-4bd9-a1b5-60782b6f7db3)

Reply:
Hello @Davidosky007  sir, It is working on my pc  sir. Can you please explain me how can I solve these . Thank you very much for your time 😁 
![Screenshot (154)](https://github.com/HtetWaiYan7191/school-library/assets/102011195/448c4862-2324-4fa4-ad06-edaf8bde852a)
![Screenshot (155)](https://github.com/HtetWaiYan7191/school-library/assets/102011195/bd0e3186-ca03-47db-b026-178e8b473e9a)
![Screenshot (156)](https://github.com/HtetWaiYan7191/school-library/assets/102011195/fb610f12-a712-4645-82e4-f40b85c80989)

Reply:
For some wired reason it didn't work on my end , kindly resubmit for review and sorry for the inconvenience thanks

Reply:
Thanks sir. I will resubmit again